couchdbkit/client.py

```diff
@@ -529,30 +536,29 @@ def save_doc(self, doc, encode_attachments=True, force_update=False,
             doc1['_attachments'] = resource.encode_attachments(doc['_attachments'])
 
         if '_id' in doc1:
-            docid = doc1['_id']
-            docid1 = resource.escape_docid(doc1['_id'])
+            docid = doc1['_id'].encode('utf-8')
+            couch_doc = Document(self.cloudant_database, docid)
+            couch_doc.update(doc1)
             try:
-                res = self.res.put(docid1, payload=doc1,
-                        **params).json_body
-            except ResourceConflict:
+                couch_doc.save()
+            except HTTPError as e:
+                if e.response.status_code != 409:
+                    raise
+
                 if force_update:
-                    doc1['_rev'] = self.get_rev(docid)
-                    res =self.res.put(docid1, payload=doc1,
-                            **params).json_body
+                    couch_doc['_rev'] = self.get_rev(docid)
+                    couch_doc.save()
                 else:
-                    raise
+                    raise ResourceConflict
+
+            res = deepcopy(couch_doc)
```

Comment:
This can be slow for large objects. Is there a more surgical method (copying less of the object) that could be used to avoid this?